### ISSUE TITLE
PainterI2V

### ISSUE THREAD

**Issue by kabachuha** [kijai/ComfyUI-WanVideoWrapper#1640]
Promises to reduce slow motion when using the lightxtv lora in Wan image2video by scaling the initial latent in a calculated way

https://github.com/princepainter/ComfyUI-PainterI2V

Tracking issue on the side https://github.com/princepainter/ComfyUI-PainterI2V/issues/5

**Comment by railep** [comment]
if motion_amplitude > 1.0:
                base_latent = concat_latent_image[:, :, 0:1]      # 首帧
                gray_latent = concat_latent_image[:, :, 1:]       # 灰帧
                
                diff = gray_latent - base_latent
                diff_mean = diff.mean(dim=(1, 3, 4), keepdim=True)
                diff_centered = diff - diff_mean
                scaled_latent = base_latent + diff_centered * motion_amplitude + diff_mean
                
                # Clamp & 组合
                scaled_latent = torch.clamp(scaled_latent, -6, 6)
                concat_latent_image = torch.cat([base_latent, scaled_latent], dim=2)
            
More difference between frames - that‘s it.

**Comment by solss-bit** [comment]
PainterLongVideo is also worth exploring. Color matching is no longer an issue. Uses previous video frames as context in subsequent video generations, while having the option to provide an initial reference image to help maintain consistency. There might be some obscure Wrapper nodes that do something similar that I'm not already aware of, and I usually just use context options, but it works really well for what it is. Native nodes don't manage memory as well as Wrapper does. Would be cool if it could be implemented somehow.

**Comment by RuneGjerde** [comment]
Looks interesting, might be worth a test, and even perhaps Kijai wants to add into the wrapper as some sort of option or setting if it works as some motion enhancer on I2V.  
For now i think it already works as it own node though https://github.com/princepainter/ComfyUI-PainterI2VforKJ 

___

A quick test: 


https://github.com/user-attachments/assets/51ed302d-86e8-4ced-8d31-bd9094a0fd66

(same seed, WanVideoWrapper I2v)

Seems to add more motions in scenes that "has motion" (car speeding down a road etc). Less change in everyday kind of scenes
But just did a quick test, not sure if there is any drawbacks like less smooth motion or something.

**Comment by RuneGjerde** [comment]
Almost feels like its a longer sequence but squeezed into same length, or something like that. The car above without the node, the camera is already moving towards same direction as with the node but never gets as far out view as with the node.   And the skateboarder below the same, with the node its almost like you see a longer video of the same 

Not sure what it does,  but works well with suitable input image  ;-) I guess the non node is a bit smoother (and slower more stable) action, while with the node you get more of the "completed motion". 

https://github.com/user-attachments/assets/1e5c66eb-35b4-4ffb-9e1b-660d7a350a3a

**Comment by RuneGjerde** [comment]
> PainterLongVideo is also worth exploring. 

Gave it a try, seems to work pretty well yeah. Maybe there are some ideas there for the context window as options. 
Ran 2-3 runs, seems to hold up pretty well depending on the source image ;-) 

Might be similar ish with VACE or some other way to give a bit more reference inputs to next "window". 

https://github.com/user-attachments/assets/52681019-bba8-45b2-a3e7-aed49f8c1dd7
